client/client.go

```diff
@@ -524,14 +524,22 @@ func (c *authenticatingClient) doAuthenticate() error {
 		err         error
 	)
 	if authDetails, err = c.authMode.Auth(c.creds); err != nil {
-		return gooseerrors.Newf(err, "authentication failed")
+		if gooseerrors.IsUnauthorised(err) {
+			return gooseerrors.NewUnauthorisedf(err, "", "authentication failed: ", c.creds)
+		} else {
+			return gooseerrors.Newf(err, "authentication failed")
+		}
 	}
 	logger := logging.FromCompat(c.logger)
 	logger.Debugf("auth details: %+v", authDetails)
 
 	c.regionServiceURLs = authDetails.RegionServiceURLs
 	if err := c.createServiceURLs(); err != nil {
-		return gooseerrors.Newf(err, "cannot create service URLs")
+		if gooseerrors.IsUnauthorised(err) {
+			return gooseerrors.NewUnauthorisedf(err, "", "failed to create URLs")
```

Comment:
Look at the createServiceURLs code, it can never return an UnauthorisedError. It never makes any api calls. It just looks at data structures and returns an error if something is wrong. So all that's needed here is a little extra context added to the error message for user consumption
return gooseerrors.Newf(err, "cannot create service URLs")